homeassistant/components/motion_blinds/cover.py

```diff
@@ -0,0 +1,259 @@
+
+
+class MotionPositionDevice(CoordinatorEntity, CoverEntity):
+    """Representation of a Motion Blind Device."""
+
+    def __init__(self, coordinator, blind, device_class, config_entry):
+        """Initialize the blind."""
+        super().__init__(coordinator)
+
+        self._blind = blind
+        self._device_class = device_class
+        self._config_entry = config_entry
+
+    @property
+    def unique_id(self):
+        """Return the unique id of the blind."""
+        return self._blind.mac
+
+    @property
+    def device_info(self):
+        """Return the device info of the blind."""
+        device_info = {
+            "identifiers": {(DOMAIN, self._blind.mac)},
+            "manufacturer": MANUFACTURER,
+            "name": f"{self._blind.blind_type}-{self._blind.mac}",
+            "model": self._blind.blind_type,
+            "via_device": (DOMAIN, self._config_entry.unique_id),
+        }
+
+        return device_info
+
+    @property
+    def name(self):
+        """Return the name of the blind."""
+        return f"{self._blind.blind_type}-{self._blind.mac}"
+
+    @property
+    def current_cover_position(self):
+        """
+        Return current position of cover.
+
+        None is unknown, 0 is open, 100 is closed.
+        """
+        if self._blind.position is None:
+            return None
+        return 100 - self._blind.position
+
+    @property
+    def device_class(self):
+        """Return the device class."""
+        return self._device_class
+
+    @property
+    def is_closed(self):
+        """Return if the cover is closed or not."""
+        return self._blind.position == 100
+
+    def open_cover(self, **kwargs):
+        """Open the cover."""
+        self._blind.Open()
+
+    def close_cover(self, **kwargs):
+        """Close cover."""
+        self._blind.Close()
+
+    def set_cover_position(self, **kwargs):
+        """Move the cover to a specific position."""
+        if ATTR_POSITION in kwargs:
+            position = kwargs[ATTR_POSITION]
+            self._blind.Set_position(100 - position)
+
+    def stop_cover(self, **kwargs):
+        """Stop the cover."""
+        self._blind.Stop()
+
+
+class MotionTiltDevice(MotionPositionDevice):
+    """Representation of a Motion Blind Device."""
+
+    @property
+    def current_cover_tilt_position(self):
+        """
+        Return current angle of cover.
+
+        None is unknown, 0 is closed/minimum tilt, 100 is fully open/maximum tilt.
+        """
+        if self._blind.angle is None:
+            return None
+        return self._blind.angle * 100 / 180
+
+    def open_cover_tilt(self, **kwargs):
+        """Open the cover tilt."""
+        self._blind.Set_angle(180)
+
+    def close_cover_tilt(self, **kwargs):
+        """Close the cover tilt."""
+        self._blind.Set_angle(0)
+
+    def set_cover_tilt_position(self, **kwargs):
+        """Move the cover tilt to a specific position."""
+        if ATTR_TILT_POSITION in kwargs:
```

Comment:
`ATTR_TITLE_POSITION` is required https://www.home-assistant.io/integrations/cover/#service-coverset_cover_tilt_position

Reply:
done